### DISCUSSION TITLE
Data is too long Error when deploying app.py

### DISCUSSION THREAD

**Discussion by JakobRied** [vercel/community#3693] [Help]
### Summary

I want to deploy my flask-app with verce. I uploaded all necessary files to my github repo and started the deployment of this repo.
However, the deployment is failing with the following error code:

**Failed to process build result for "app.py". Data: {"type":"Lambda"}.
Error: data is too long**

I could not find any information about this Error message online. What could cause this problem? 
You can find the code in my repo: (https://github.com/JakobRied/TaxiDetection)

### Steps to Reproduce

The only thing I did:
1. Connect Github Repo to Vercel
2. Deploy the Repo with standard settings

**Comment by sumersao** [comment]
I have this issue too!

**Comment by mirokery** [comment]
same here with fastapi if I want to deploy just dummy hello word app it works when I want deploy app for object detection it fails on
Installing required dependencies...

this is my req.txt:
fastapi[all]
uvicorn
ultralytics
opencv-python
Pillow

**Comment by rjrobben** [comment]
same here, guessing if it's due to the large file on github like .pt checkpoint files?

**Comment by AndyMik90** [comment]
Also getting "Error: data is too long", works with hello world deployment of FastAPI, but when switching to my own python script it fails.

**Comment by mirokery** [comment]
what others free platforms did you try ? vercel doesn't work for me because of this ...also replit doesn't work for me...where I was able to deploy was render... you can try that

**Comment by Wolfleader101** [comment]
bump to this - getting same issue and not sure how to fix it

**Reply by Nithin1729S** [reply]
Did you find any alternative ? Please share

**Comment by pj-13-shukla** [comment]
same issue

**Reply by Nithin1729S** [reply]
Did you find any alternative ? Please share

**Comment by oguzhantasimaz** [comment]
Same issue with fastapi, here are my requirements

```
upstash_vector==0.2.1
Pillow==10.2.0
torch==2.1.2
transformers==4.36.2
torchvision==0.16.2
numpy==1.26.2
slowapi==0.1.9
openai==1.10.0
uvicorn==0.25.0
fastapi==0.109.2
typing==3.7.4.3
python-multipart==0.0.6
```

**Reply by rjmaier7** [reply]
Same issue here. 
My requirements are:
```
fastapi==0.100.1
uvicorn[standard]==0.23.2
supabase==2.4.2
pydantic==2.6.4
pydantic_core==2.16.3
torch==2.2.2
torchvision==0.17.2
pandas==2.2.2
numpy==1.26.4
```

**Reply by Nithin1729S** [reply]
Did you find any alternative ? Please share

**Comment by twentytwokhz** [comment]
I would dare say it most likely is related to a configuration issue.
I changed my `vercel.json` config to include the `maxLambdaSize` property:

```json
{
  "version": 2,
  "builds": [
    {
      "use": "@vercel/python",
      "config": {
        "maxLambdaSize": "450mb"
      }
    }
  ],
  "rewrites": [{ "source": "/(.*)", "destination": "/api/index" }]
}

```

I will get back with my results

**Reply by millivan** [reply]
Hey, just curious how are your results? Coincidentally I'm having this issue right now

**Reply by FoxTail389** [reply]
Same, it's funny how we have the same problem at the same time

**Reply by twentytwokhz** [reply]
Same issue

```log
22:29:01.382 | Deploying outputs...
22:33:30.292 | Failed to process build result for "api/index.py". Data: {"type":"Lambda"}.
22:33:30.298 | Error: data is too long
```

If I could only find out the outputs size... even though I set ithe `"maxLambdaSize": "2000mb"` 🤔

**Reply by FoxTail389** [reply]
Any other thoughts?

**Reply by millivan** [reply]
I tried asking my friends, and so far the easy answer is Vercel is not particularly good at handling Python backends, especially since I used scikit-learn and it seems that the library is too large.

**Reply by FoxTail389** [reply]
Do you know any better free alternatives?

**Reply by millivan** [reply]
He suggested Railway, they have a free trial. I haven't really explored it yet

**Reply by twentytwokhz** [reply]
Alternatives depend on your particular usecase: api backend, inference model or simple frontend
- Replicate
- Hyperstack
- Beam
- Dedicated spaces in HuggingFace

**Comment by PrabirKalwani** [comment]
can you guys suggest any alternatives for vercel since i want to deploy large flask websites with custom llms

**Reply by adeola2310** [reply]
i am guessing PythonAnywhere might work for what you are trying to achieve

**Reply by Nithin1729S** [reply]
Did you find any alternative ? Please share

**Reply by PrabirKalwani** [reply]
Im still looking out for alternatives for the time being i used AWS lambda which is also a server less function

**Comment by JUSTSUJAY** [comment]
I also have a trained RF learning agent via neural network(pytorch settings) so I want to deploy it to make it interactive so others can interact with it too. Gettting Data is too long error, 

if anyone has any (free hosting)suggestions, please let me know

**Reply by Nithin1729S** [reply]
Did you find any alternative ? Please share
